### PR TITLE
Add PAPI 7.0.1 and 7.1.0

### PR DESCRIPTION
**Pull request by wyphan**
Supersedes #33940

SHA256 sum computed manually after verifying the MD5 sum with the one posted at the ICL website.

### CODE REVIEW

File: var/spack/repos/builtin/packages/papi/package.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class Papi(AutotoolsPackage, ROCmPackage):
     git = "https://github.com/icl-utk-edu/papi"
 
     version("master", branch="master")
+    version("7.0.1", sha256="c105da5d8fea7b113b0741a943d467a06c98db959ce71bdd9a50b9f03eecc43e")
```
Comment (naromero77) [inline var/spack/repos/builtin/packages/papi/package.py:35]: Thanks for the PR! Could you please add a comment right below 7.0.1, says 7.0.0 was intentionally omitted due to build issues or something like that.

**Review by naromero77 — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/papi/package.py (hunk 1 above)
Comment (wyphan) [reply var/spack/repos/builtin/packages/papi/package.py:35]: Done!

**Comment by wyphan** [comment]
Confirmed working on my dev rig (AMD Zen 2, NVIDIA "Ampere" 8.6, AMD`gfx900`, Ubuntu 20.04.6):
```plain
wyp@basecamp:~/work/spack$ spack find
==> In environment test
==> Root specs
papi@7.0.1 ~cuda~nvml~rocm~rocm_smi  papi@7.0.1 +cuda+nvml~rocm~rocm_smi
papi@7.0.1 ~cuda~nvml+rocm+rocm_smi  papi@7.0.1 +cuda+nvml+rocm+rocm_smi

wyp@basecamp:~/work/spack$ spack install
:
[ install succeeded ]

wyp@basecamp:~/work/spack$ spack load papi +cuda +rocm

wyp@basecamp:~/work/spack$ papi_avail -a
Available PAPI preset and user defined events plus hardware information.
--------------------------------------------------------------------------------
PAPI version             : 7.0.1.0
Operating system         : Linux 5.15.0-75-generic
Vendor string and code   : AuthenticAMD (2, 0x2)
Model string and code    : AMD Ryzen 5 3600X 6-Core Processor (113, 0x71)
CPU revision             : 0.000000
CPUID                    : Family/Model/Stepping 23/113/0, 0x17/0x71/0x00
CPU Max MHz              : 4000
CPU Min MHz              : 2200
Total cores              : 12
SMT threads per core     : 2
Cores per socket         : 6
Sockets                  : 1
Cores per NUMA region    : 12
NUMA regions             : 1
Running in a VM          : no
Number Hardware Counters : 5
Max Multiplex Counters   : 384
Fast counter read (rdpmc): yes
--------------------------------------------------------------------------------

================================================================================
  PAPI Preset Events
================================================================================
    Name        Code    Deriv Description (Note)

PAPI_L1_ICM  0x80000001  No   Level 1 instruction cache misses
PAPI_L2_DCM  0x80000002  No   Level 2 data cache misses
PAPI_L2_ICM  0x80000003  No   Level 2 instruction cache misses
PAPI_TLB_DM  0x80000014  No   Data translation lookaside buffer misses
PAPI_TLB_IM  0x80000015  Yes  Instruction translation lookaside buffer misses
PAPI_BR_TKN  0x8000002c  No   Conditional branch instructions taken
PAPI_BR_MSP  0x8000002e  No   Conditional branch instructions mispredicted
PAPI_TOT_INS 0x80000032  No   Instructions completed
PAPI_FP_INS  0x80000034  No   Floating point instructions
PAPI_BR_INS  0x80000037  No   Branch instructions
PAPI_TOT_CYC 0x8000003b  No   Total cycles
PAPI_L2_DCH  0x8000003f  No   Level 2 data cache hits
PAPI_L1_DCA  0x80000040  No   Level 1 data cache accesses
PAPI_L2_DCR  0x80000044  No   Level 2 data cache reads
PAPI_L2_ICH  0x8000004a  No   Level 2 instruction cache hits
PAPI_L2_ICR  0x80000050  No   Level 2 instruction cache reads
PAPI_FP_OPS  0x80000066  No   Floating point operations
--------------------------------------------------------------------------------
Of 17 available events, 1 is derived.

wyp@basecamp:~/work/spack$ export PAPI_ROCMSMI_ROOT=/opt/rocm-5.4.3/rocm_smi

wyp@basecamp:~/work/spack$ papi_component_avail 
Available components and hardware information.
--------------------------------------------------------------------------------
PAPI version             : 7.0.1.0
Operating system         : Linux 5.15.0-75-generic
Vendor string and code   : AuthenticAMD (2, 0x2)
Model string and code    : AMD Ryzen 5 3600X 6-Core Processor (113, 0x71)
CPU revision             : 0.000000
CPUID                    : Family/Model/Stepping 23/113/0, 0x17/0x71/0x00
CPU Max MHz              : 4000
CPU Min MHz              : 2200
Total cores              : 12
SMT threads per core     : 2
Cores per socket         : 6
Sockets                  : 1
Cores per NUMA region    : 12
NUMA regions             : 1
Running in a VM          : no
Number Hardware Counters : 5
Max Multiplex Counters   : 384
Fast counter read (rdpmc): yes
--------------------------------------------------------------------------------

Compiled-in components:
Name:   perf_event              Linux perf_event CPU counters
Name:   perf_event_uncore       Linux perf_event CPU uncore and northbridge
   \-> Disabled: Insufficient permissions for uncore access.  Set /proc/sys/kernel/perf_event_paranoid to 0 or run as root.
Name:   example                 A simple example component
Name:   cuda                    CUDA events and metrics via NVIDIA CuPTI interfaces
Name:   nvml                    NVML provides the API for monitoring NVIDIA hardware (power usage, temperature, fan speed, etc)
Name:   rocm                    GPU events and metrics via AMD ROCm-PL API
Name:   rocm_smi                AMD GPU System Management Interface via rocm_smi_lib
   \-> Disabled: Error while initializing the native event table.
Name:   sysdetect               System info detection component

Active components:
Name:   perf_event              Linux perf_event CPU counters
                                Native: 141, Preset: 17, Counters: 5
                                PMUs supported: perf, perf_raw, amd64_fam17h_zen2

Name:   example                 A simple example component
                                Native: 4, Preset: 0, Counters: 3

Name:   cuda                    CUDA events and metrics via NVIDIA CuPTI interfaces
                                Native: 259254, Preset: 0, Counters: 259254

Name:   nvml                    NVML provides the API for monitoring NVIDIA hardware (power usage, temperature, fan speed, etc)
                                Native: 28, Preset: 0, Counters: 28

Name:   rocm                    GPU events and metrics via AMD ROCm-PL API
                                Native: 0, Preset: 0, Counters: 0

Name:   sysdetect               System info detection component
                                Native: 0, Preset: 0, Counters: 0
--------------------------------------------------------------------------------
```

**Review by naromero77 — DISMISSED** [review]
Thanks!

**Comment by naromero77** [comment]
There is a failure that seems to be unrelated to this patch, going to restart the pipeline.

**Comment by naromero77** [comment]
@spackbot run pipeline

**Comment by eugeneswalker** [comment]
[(specs) papi/jjh2oak 7.0.1 cce@=15.0.1 linux-rhel8-zen4 E4S Cray](https://gitlab.spack.io/spack/spack/-/jobs/7405712)
```
...
ftn-78 ftn: ERROR in command line
  The -f option has an invalid argument, "ree".
make[2]: *** [Makefile:46: query_device_simple_f] Error 1
make[2]: Leaving directory '/tmp/gitlab-runner-1/spack-stage/spack-stage-papi-7.0.1-jjh2oakmkillg524lzt2eo2heg6yqet6/spack-src/src/components/sysdetect/tests'
make[1]: *** [Makefile.inc:265: comp_tests] Error 2
make[1]: Leaving directory '/tmp/gitlab-runner-1/spack-stage/spack-stage-papi-7.0.1-jjh2oakmkillg524lzt2eo2heg6yqet6/spack-src/src'
make: *** [Rules.pfm4_pe:45: libpfm4/lib/libpfm.a] Error 2
==> Error: ProcessError: Command exited with status 2:
    'make' '-j1' 'V=1'
3 errors found in build log:
     1334    make[2]: Entering directory '/tmp/gitlab-runner-1/spack-stage/spac
             k-stage-papi-7.0.1-jjh2oakmkillg524lzt2eo2heg6yqet6/spack-src/src/
             components/sysdetect/tests'
     1335    /home/gitlab-runner-1/builds/Q2MuKA89/0/spack/spack/lib/spack/env/
             cce/cc -g  -DPAPI_NUM_COMP=4 -O2 -I. -I../../.. -I../../../testlib
              -I../../../validation_tests -I/home/gitlab-runner-1/builds/Q2MuKA
             89/0/spack/spack/opt/spack/__spack_path_placeholder__/__spack_path
             _placeholder__/__spack_path_placeholder__/__spack_path_placeholder
             __/__spack_path_placeholder__/__spack_path_placeholder__/__spack_p
             ath_placeholder__/__spa/morepadding/linux-rhel8-zen4/cce-15.0.1/pa
             pi-7.0.1-jjh2oakmkillg524lzt2eo2heg6yqet6/include -c -o query_devi
             ce_simple.o query_device_simple.c
     1336    /home/gitlab-runner-1/builds/Q2MuKA89/0/spack/spack/lib/spack/env/
             cce/cc -g  -DPAPI_NUM_COMP=4 -I. -I../../.. -I../../../testlib -I.
             ./../../validation_tests -I/home/gitlab-runner-1/builds/Q2MuKA89/0
             /spack/spack/opt/spack/__spack_path_placeholder__/__spack_path_pla
             ceholder__/__spack_path_placeholder__/__spack_path_placeholder__/_
             _spack_path_placeholder__/__spack_path_placeholder__/__spack_path_
             placeholder__/__spa/morepadding/linux-rhel8-zen4/cce-15.0.1/papi-7
             .0.1-jjh2oakmkillg524lzt2eo2heg6yqet6/include -o query_device_simp
             le query_device_simple.o ../../../testlib/libtestlib.a ../../../li
             bpapi.a
     1337    /home/gitlab-runner-1/builds/Q2MuKA89/0/spack/spack/lib/spack/env/
             cce/ftn -free -I../../.. -o query_device_simple_f query_device_sim
             ple_f.F ../../../libpapi.a
     1338    ftn-78 ftn: ERROR in command line
     1339      The -f option has an invalid argument, "ree".
  >> 1340    make[2]: *** [Makefile:46: query_device_simple_f] Error 1
     1341    make[2]: Leaving directory '/tmp/gitlab-runner-1/spack-stage/spack
             -stage-papi-7.0.1-jjh2oakmkillg524lzt2eo2heg6yqet6/spack-src/src/c
             omponents/sysdetect/tests'
  >> 1342    make[1]: *** [Makefile.inc:265: comp_tests] Error 2
     1343    make[1]: Leaving directory '/tmp/gitlab-runner-1/spack-stage/spack
             -stage-papi-7.0.1-jjh2oakmkillg524lzt2eo2heg6yqet6/spack-src/src'
  >> 1344    make: *** [Rules.pfm4_pe:45: libpfm4/lib/libpfm.a] Error 2
```

**Comment by G-Ragghianti** [comment]
I have posted an issue on this to papi upstream: https://github.com/icl-utk-edu/papi/issues/24

Do we want to maybe include a patch changing FFLAGS to something reasonable for CCE/FTN to get this going within spack?

**Comment by gcongiu** [comment]
@wyphan I have reflected your changes in PAPI (see above reference). Could you please check the commits and your user details are correct? Thank you

**Comment by wyphan** [comment]
@gcongiu Looks good to me! I'm gonna restart the CI pipeline since they seem to be general CI failures.

**Comment by wyphan** [comment]
@spackbot run pipeline

**Review by naromero77 — DISMISSED** [review]
Thank's for the Cray patch.

**Comment by eugeneswalker** [comment]
Getting close I think. Just one error left with `papi+rocm`
* [(specs) papi/vsdrwml 7.0.1 gcc@=11.1.0 linux-ubuntu20.04-x86_64_v3 E4S](https://gitlab.spack.io/spack/spack/-/jobs/7424607)

```
d.lld: error: unable to find library -lomp
clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [<builtin>: sample_single_kernel_monitoring] Error 1
make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-papi-7.0.1-vsdrwmlh7ilwvanrib2jdlxanri3uqqy/spack-src/src/components/rocm/tests'
make[1]: *** [Makefile.inc:265: comp_tests] Error 2
make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-papi-7.0.1-vsdrwmlh7ilwvanrib2jdlxanri3uqqy/spack-src/src'
make: *** [Rules.pfm4_pe:43: libpfm4/lib/libpfm.a] Error 2
==> Error: ProcessError: Command exited with status 2:
    'make' '-j1' 'V=1'
5 errors found in build log:
     1291    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-papi
             -7.0.1-vsdrwmlh7ilwvanrib2jdlxanri3uqqy/spack-src/src/components/e
             xample/tests'
     1292    + make -C components/rocm/tests
     1293    make[2]: Entering directory '/tmp/root/spack-stage/spack-stage-pap
             i-7.0.1-vsdrwmlh7ilwvanrib2jdlxanri3uqqy/spack-src/src/components/
             rocm/tests'
     1294    /home/software/spack/__spack_path_placeholder__/__spack_path_place
             holder__/__spack_path_placeholder__/__spack_path_placeholder__/__s
             pack_path_placeholder__/__spack_path_placeholder__/__spack_path_pl
             aceholder__/__spack_path_placeholder__/__spack_path_placeh/linux-u
             buntu20.04-x86_64_v3/gcc-11.1.0/hip-5.4.3-ltqcijtgyd4blvcvmwvbjmpu
             fsl5c7h2/bin/hipcc -g -O0 -fopenmp -I/home/software/spack/__spack_
             path_placeholder__/__spack_path_placeholder__/__spack_path_placeho
             lder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa
             ck_path_placeholder__/__spack_path_placeholder__/__spack_path_plac
             eholder__/__spack_path_placeh/linux-ubuntu20.04-x86_64_v3/gcc-11.1
             .0/hsa-rocr-dev-5.4.3-copkdm76p3szvu4urdvzvan2rrzuwthe/include -I/
             home/software/spack/__spack_path_placeholder__/__spack_path_placeh
             older__/__spack_path_placeholder__/__spack_path_placeholder__/__sp
             ack_path_placeholder__/__spack_path_placeholder__/__spack_path_pla
             ceholder__/__spack_path_placeholder__/__spack_path_placeh/linux-ub
             untu20.04-x86_64_v3/gcc-11.1.0/hsa-rocr-dev-5.4.3-copkdm76p3szvu4u
             rdvzvan2rrzuwthe/include/hip -I/home/software/spack/__spack_path_p
             laceholder__/__spack_path_placeholder__/__spack_path_placeholder__
             /__spack_path_placeholder__/__spack_path_placeholder__/__spack_pat
             h_placeholder__/__spack_path_placeholder__/__spack_path_placeholde
             r__/__spack_path_placeh/linux-ubuntu20.04-x86_64_v3/gcc-11.1.0/hsa
             -rocr-dev-5.4.3-copkdm76p3szvu4urdvzvan2rrzuwthe/include/hsa -I/ho
             me/software/spack/__spack_path_placeholder__/__spack_path_placehol
             der__/__spack_path_placeholder__/__spack_path_placeholder__/__spac
             k_path_placeholder__/__spack_path_placeholder__/__spack_path_place
             holder__/__spack_path_placeholder__/__spack_path_placeh/linux-ubun
             tu20.04-x86_64_v3/gcc-11.1.0/hsa-rocr-dev-5.4.3-copkdm76p3szvu4urd
             vzvan2rrzuwthe/hip/include/hip -I/home/software/spack/__spack_path
             _placeholder__/__spack_path_placeholder__/__spack_path_placeholder
             __/__spack_path_placeholder__/__spack_path_placeholder__/__spack_p
             ath_placeholder__/__spack_path_placeholder__/__spack_path_placehol
             der__/__spack_path_placeh/linux-ubuntu20.04-x86_64_v3/gcc-11.1.0/h
             sa-rocr-dev-5.4.3-copkdm76p3szvu4urdvzvan2rrzuwthe/hsa/include/hsa
              -I. -I../../.. -I../../../testlib -I../../../validation_tests -I/
             home/software/spack/__spack_path_placeholder__/__spack_path_placeh
             older__/__spack_path_placeholder__/__spack_path_placeholder__/__sp
             ack_path_placeholder__/__spack_path_placeholder__/__spack_path_pla
             ceholder__/__spack_path_placeholder__/__spack_path_placeh/morepadd
             ing/linux-ubuntu20.04-x86_64_v3/gcc-11.1.0/papi-7.0.1-vsdrwmlh7ilw
             vanrib2jdlxanri3uqqy/include  -c -o sample_single_kernel_monitorin
             g.o sample_single_kernel_monitoring.cpp
     1295    /home/software/spack/__spack_path_placeholder__/__spack_path_place
             holder__/__spack_path_placeholder__/__spack_path_placeholder__/__s
             pack_path_placeholder__/__spack_path_placeholder__/__spack_path_pl
             aceholder__/__spack_path_placeholder__/__spack_path_placeh/linux-u
             buntu20.04-x86_64_v3/gcc-11.1.0/hip-5.4.3-ltqcijtgyd4blvcvmwvbjmpu
             fsl5c7h2/bin/hipcc -g -O0 -fopenmp -I/home/software/spack/__spack_
             path_placeholder__/__spack_path_placeholder__/__spack_path_placeho
             lder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa
             ck_path_placeholder__/__spack_path_placeholder__/__spack_path_plac
             eholder__/__spack_path_placeh/linux-ubuntu20.04-x86_64_v3/gcc-11.1
             .0/hsa-rocr-dev-5.4.3-copkdm76p3szvu4urdvzvan2rrzuwthe/include -I/
             home/software/spack/__spack_path_placeholder__/__spack_path_placeh
             older__/__spack_path_placeholder__/__spack_path_placeholder__/__sp
             ack_path_placeholder__/__spack_path_placeholder__/__spack_path_pla
             ceholder__/__spack_path_placeholder__/__spack_path_placeh/linux-ub
             untu20.04-x86_64_v3/gcc-11.1.0/hsa-rocr-dev-5.4.3-copkdm76p3szvu4u
             rdvzvan2rrzuwthe/include/hip -I/home/software/spack/__spack_path_p
             laceholder__/__spack_path_placeholder__/__spack_path_placeholder__
             /__spack_path_placeholder__/__spack_path_placeholder__/__spack_pat
             h_placeholder__/__spack_path_placeholder__/__spack_path_placeholde
             r__/__spack_path_placeh/linux-ubuntu20.04-x86_64_v3/gcc-11.1.0/hsa
             -rocr-dev-5.4.3-copkdm76p3szvu4urdvzvan2rrzuwthe/include/hsa -I/ho
             me/software/spack/__spack_path_placeholder__/__spack_path_placehol
             der__/__spack_path_placeholder__/__spack_path_placeholder__/__spac
             k_path_placeholder__/__spack_path_placeholder__/__spack_path_place
             holder__/__spack_path_placeholder__/__spack_path_placeh/linux-ubun
             tu20.04-x86_64_v3/gcc-11.1.0/hsa-rocr-dev-5.4.3-copkdm76p3szvu4urd
             vzvan2rrzuwthe/hip/include/hip -I/home/software/spack/__spack_path
             _placeholder__/__spack_path_placeholder__/__spack_path_placeholder
             __/__spack_path_placeholder__/__spack_path_placeholder__/__spack_p
             ath_placeholder__/__spack_path_placeholder__/__spack_path_placehol
             der__/__spack_path_placeh/linux-ubuntu20.04-x86_64_v3/gcc-11.1.0/h
             sa-rocr-dev-5.4.3-copkdm76p3szvu4urdvzvan2rrzuwthe/hsa/include/hsa
              -I. -I../../.. -I../../../testlib -I../../../validation_tests -I/
             home/software/spack/__spack_path_placeholder__/__spack_path_placeh
             older__/__spack_path_placeholder__/__spack_path_placeholder__/__sp
             ack_path_placeholder__/__spack_path_placeholder__/__spack_path_pla
             ceholder__/__spack_path_placeholder__/__spack_path_placeh/morepadd
             ing/linux-ubuntu20.04-x86_64_v3/gcc-11.1.0/papi-7.0.1-vsdrwmlh7ilw
             vanrib2jdlxanri3uqqy/include  -c -o matmul.o matmul.cpp
     1296    /home/software/spack/__spack_path_placeholder__/__spack_path_place
             holder__/__spack_path_placeholder__/__spack_path_placeholder__/__s
             pack_path_placeholder__/__spack_path_placeholder__/__spack_path_pl
             aceholder__/__spack_path_placeholder__/__spack_path_placeh/linux-u
             buntu20.04-x86_64_v3/gcc-11.1.0/hip-5.4.3-ltqcijtgyd4blvcvmwvbjmpu
             fsl5c7h2/bin/hipcc -ldl ../../../libpapi.a ../../../testlib/libtes
             tlib.a -fopenmp  sample_single_kernel_monitoring.o matmul.o   -o s
             ample_single_kernel_monitoring
  >> 1297    ld.lld: error: unable to find library -lomp
  >> 1298    clang-15: error: linker command failed with exit code 1 (use -v to
              see invocation)
  >> 1299    make[2]: *** [<builtin>: sample_single_kernel_monitoring] Error 1
     1300    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-papi
             -7.0.1-vsdrwmlh7ilwvanrib2jdlxanri3uqqy/spack-src/src/components/r
             ocm/tests'
  >> 1301    make[1]: *** [Makefile.inc:265: comp_tests] Error 2
     1302    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-papi
             -7.0.1-vsdrwmlh7ilwvanrib2jdlxanri3uqqy/spack-src/src'
  >> 1303    make: *** [Rules.pfm4_pe:43: libpfm4/lib/libpfm.a] Error 2
```

**Review by eugeneswalker — DISMISSED** [review]

**Comment by wyphan** [comment]
Oh well, now it's llvm-amdgpu that's acting up.
https://gitlab.spack.io/spack/spack/-/jobs/7429562#L885

**Comment by naromero77** [comment]
My suggestion would be to set conflicts for combinations (variants) that are not working and report them to the upstream developers. Set enough conflicts so that the CI test pass for now. If the PAPI developers fix some of these issues, we can always patch 7.0.1 from upstream.

I don't want this PR to become a huge effort for @wyphan.

**Comment by wyphan** [comment]
Well, it's specifically `papi +rocm` that I wanted to deploy, so I guess I need to figure out a workaround for this

**Comment by naromero77** [comment]
@spackbot run pipeline

**Comment by wyphan** [comment]
Almost there, just need to inject the install dir from the regular `llvm` build into the CMake configure for `llvm-amdgpu` :joy:
Edit: thanks @kwryankrattiger for the pointer, using `rocm-openmp-extras` instead.

**Comment by wyphan** [comment]
@srekolam @renjithravindrankannath if you inspect the CI failures above, it's a concretization error. I tried to reproduce it using `spack ci reproduce-build`, turns out it involves `hip@master` combined with `rocprofiler-dev@4.5.2` and so on... I think this needs to be fixed at the `ROCmPackage` build system level instead of the `llvm-amdgpu` package level

**Comment by naromero77** [comment]
@wyphan The upstream only has PAPI 6.x which is almost three years old. I guess there were issues with the LLVM-AMDGPU package that held up this PR. One possibility is to just include PAPI 7.0.1 and 7.1.0 in a single PR and fix the LLVM-AMDGPU in a different set of PRs.

**Comment by wyphan** [comment]
@naromero77 Yeah that sounds reasonable. I'll try to finalize the PR based on this plan later today.

**Review by naromero77 — DISMISSED** [review]
Thanks!

**Comment by wyphan** [comment]
Ugh. Looks like the SHA sum for the tarball from ICL website and the one in GitHub release tag are different. Fixed now to use the ICL website's tarball.

**Review by naromero77 — APPROVED** [review]
LGTM. 

I will merge tomorrow if I don't hear back from @G-Ragghianti or the others.

**Comment by haampie** [comment]
Hi @wyphan, Gitlab CI didn't report this issue with the PR: https://gitlab.spack.io/spack/spack/-/jobs/9653923

> /home/gitlab-runner-protected-3/builds/Cj9btwqn/0/spack/spack/lib/spack/env/cce/ftn -ffree-form -ffree -I../../.. -o query_device_simple_f query_device_simple_f.F ../../../libpapi.a 
> ftn-78 ftn: ERROR in command line
>   The -f option has an invalid argument, "free-form".

Do you wanna take a look at that?

**Comment by haampie** [comment]
Notice you don't have to create patch files, you can just set `make VARIABLE=VALUE`

**Comment by wyphan** [comment]
Huh, I thought that CCE patch already got merged into the main codebase before the 7.1.0 release, so I only applied it for 7.0.1 in the recipe.

**Pull request merged**